.github/workflows/stargate.yml

```diff
@@ -0,0 +1,34 @@
+name: Stargate
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install dependencies
+      run: ./scripts/debian_deps.sh
+    - name: Update git submodules
+      run: git submodule init && git submodule update
+
+    - name: Build
+      run: ./scripts/deb.py && ./scripts/rpm.py --nodeps
```

Comment:
Will `rpm.py` work on Ubuntu?

Reply:
...and also, will the resulting RPM work on other RPM distros?  If building from a different OS, it's better to use an older distro with older glibc.  Running an executable linked to a older glibc on a newer glibc works, but linking to a newer glibc and then running on an older glibc does not.  Ubuntu 20.04 or Ubuntu 18.04 would be better.  Or if we can have separate jobs for ubuntu-latest, fedora-latest and debian-latest, that would also work.

Reply:
I guess `ubuntu-latest` actually means 20.04, ie: latest LTS.  But it would still be better to target an older GCC and glibc like Ubuntu 18.04 so that it will work on older distros, since when Github 
Actions bumps ubuntu-latest to 22.04, then the packages would no longer work on most older distros.  Also, `deb.py` and `rpm.py` will probably name the packages something like:
```
stargate-21.11.9-ubuntu-20.04.rpm
stargate-21.11.9-ubuntu-20.04.deb
```
when they should just be something like:
```
stargate-21.11.9-generic.rpm
stargate-21.11.9-generic.deb
```

Maybe add CLI args  like:
```
./rpm.py --os-name=generic
./deb.py --os-name=generic
```
, where if --os-name is not specified, it automatically detects it as it does now

Reply:
> Will `rpm.py` work on Ubuntu?

Yes. I tested it [on my fork](https://github.com/pasqualirb/stargate/actions).

> ...and also, will the resulting RPM work on other RPM distros?

I've successfully installed the package on Fedora 35.

> If building from a different OS, it's better to use an older distro with older glibc. Running an executable linked to a older glibc on a newer glibc works, but linking to a newer glibc and then running on an older glibc does not. Ubuntu 20.04 or Ubuntu 18.04 would be better. Or if we can have separate jobs for ubuntu-latest, fedora-latest and debian-latest, that would also work.
> I guess ubuntu-latest actually means 20.04, ie: latest LTS. But it would still be better to target an older GCC and glibc like Ubuntu 18.04 so that it will work on older distros, since when Github
Actions bumps ubuntu-latest to 22.04, then the packages would no longer work on most older distros.

Agreed. I added 3df3269d5b09e99351cefda4fc1203aeefac2d1b targetting Ubuntu 18.04

Fedora [doesn't seem to be available as an image yet](https://github.com/actions/virtual-environments/tree/ubuntu20/20211101.1#available-environments).

> Also, deb.py and rpm.py will probably name the packages something like:
> ...

RPM output is generated by `rpmbuild` and it's already without distro information. E.g.: `stargate-21.11.1-1.x86_64.rpm`. 

For DEB package, I added 04d9b289e52a6da9793139da91d23ac71fb48a9b which removes explicit filename definition and uses `dpkg-name` for renaming the package as described in [`dpkg-name` manual page](https://manpages.debian.org/bullseye/dpkg-dev/dpkg-name.1.en.html):

> This manual page documents the dpkg-name program which provides an easy way to rename Debian packages into their full package names. A full package name consists of **package_version_architecture.package-type** as specified in the control file of the package.